src/frontend/src/pages/FinancePage/ReimbursementRequestsSection.tsx

```diff
@@ -26,13 +34,79 @@ const ReimbursementRequestTable = ({
   const displayedReimbursementRequests =
     tabValue === 1 && allReimbursementRequests ? allReimbursementRequests : userReimbursementRequests;
 
-  const rows = displayedReimbursementRequests
-    .map(createReimbursementRequestRowData)
-    .sort((a, b) => (a.dateSubmitted > b.dateSubmitted ? -1 : 1));
+  // function getComparator<Key extends keyof any>(
+  //   order: Order,
+  //   orderBy: Key
+  // ): (a: { [key in Key]: number | string }, b: { [key in Key]: number | string }) => number {
+  //   return order === 'desc' ? (a, b) => descendingComparator(a, b, orderBy) : (a, b) => -descendingComparator(a, b, orderBy);
+  // }
+
+  function descendingComparator<T>(a: T, b: T, orderBy: keyof T) {
+    if (b[orderBy] < a[orderBy]) {
+      return -1;
+    }
+    if (b[orderBy] > a[orderBy]) {
+      return 1;
+    }
+    return 0;
+  }
+
+  function statusDescendingComparator(a: ReimbursementStatusType, b: ReimbursementStatusType): number {
+    const statusOrder = new Map<ReimbursementStatusType, number>([
+      [ReimbursementStatusType.PENDING_FINANCE, 1],
+      [ReimbursementStatusType.SABO_SUBMITTED, 2],
+      [ReimbursementStatusType.ADVISOR_APPROVED, 3],
+      [ReimbursementStatusType.DENIED, 5]
+    ]);
+
+    let bConverted = statusOrder.get(b);
+    let aConverted = statusOrder.get(a);
+
+    if (bConverted !== undefined && aConverted !== undefined) {
+      if (bConverted < aConverted) {
+        return -1;
+      }
+      if (bConverted > aConverted) {
+        return 1;
+      }
+    }
+    return 0;
+  }
+
+  function vendorDescendingComparator(a: Vendor, b: Vendor): number {
+    if (b.name < a.name) {
+      return -1;
+    }
+    if (b.name > a.name) {
+      return 1;
+    }
+    return 0;
+  }
+
+  const rows = displayedReimbursementRequests.map(createReimbursementRequestRowData).sort((a, b) => {
```

Comment:
One question for you here: I might have missed it but how did you manage to sort these `rows` consistently whenever the user clicks on one of the table head labels (Amount, vendor, etc). I checked the sorting logic in my localhost and saw that it worked fine, but I wasn't able to find a place in the code where we kept updating the order of rows consistently. 

In other words, from MUI doc, I saw that they used React's `useMemo` hook to consistently update the order of the rows. 

Let me know your thoughts since I'm curious about it!

Reply:
I think it said in the MUI doc that useMemo is for outdated browsers or something and that modern browsers support Array.sort (since 2020). The way I sorted is by using comparators (descendingComparator function, statusDescendingComparator (which is how I order the status by Reid's request), and vendorDescendingComparator) which are inputted into the .sort on line 86. The direction of the sort and the column of the sort is stored in orderBy and order. I then negate the comparator based on the direction.